logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java

```diff
@@ -223,6 +243,32 @@ byte[] pollEntryBytes(long timeout) throws IOException, InterruptedException {
         return event;
     }
 
+    /**
+     * Acknowledge last read event, must match every {@code #pollEntry} call.
+     * */
+    public void markForDelete() {
+        if (!cleanConsumed) {
+            // ack-event is useful only when clean consumed is enabled.
+            return;
+        }
+        if (lastConsumedReader == null) {
+            // no reader to a consumed segment is present
+            return;
+        }
+
+        segmentCallback.segmentCompleted();
```

Comment:
Is a segment only considered "completed" when `cleanConsumed` is set? Or would we want to call this whenever a segment has finished reading?

Reply:
A segment is considered "completed" only when the `cleanConsumed` feature flag is enabled. This is a double check, the DLQ input plugin should call the method `markForDelete` only when the `cleanConsumed` setting, so it should never happen that `markForDelete` is invoked when the flag is disabled, but this is a sort of API and we need to check the contract.